blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java

```diff
@@ -84,10 +84,7 @@ public Integer getWeatherScore() {
 
     @Override
     public BlueRun getLatestRun() {
-        if(job.getLastBuild() == null){
-            return null;
-        }
-        return AbstractRunImpl.getBlueRun(job.getLastBuild(), this);
+        return getRuns().iterator().next();
```

Comment:
I think I missed the fact that BluePipeline.getRuns() gives merge of queued and active runs. I still think there is bug where if no job has been run, it might result in to `NoSuchElementException`.